### ISSUE TITLE
`app_deactivate` and `app_activate` hooks are called in reverse order

### ISSUE THREAD

**Issue by ekiefl** [talonvoice/talon#77]
Currently teardown operations from leaving an application have the potential to overwrite buildup operations from opening an application.

This can be checked the following script:

```
from talon import ui

def on_activate(app):
    print('{0.name} is activated'.format(app))
def on_deactivate(app):
    print('{0.name} is deactivated'.format(app))

ui.register('app_deactivate', on_deactivate)
ui.register('app_activate', on_activate)
```

```
2018-12-10 15:29:30    IO Google Chrome is activated
2018-12-10 15:29:30    IO TextEdit is deactivated
```

**Comment by lunixbochs** [comment]
fyi python has fstrings, like this: `print(f"{app.name} is activated")`

**Comment by lunixbochs** [comment]
fixed in 0.0.8.21
